projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1776,6 +1783,36 @@ public function has_paid_plan( $refresh_from_wpcom = false ) {
 		}
 		return $has_paid_plan;
 	}
+
+	/**
+	 * Check if we have a Jetpack Social Advanced plan.
+	 *
+	 * @param bool $refresh_from_wpcom Whether to force refresh the plan check.
+	 *
+	 * @return bool True if we have an Advanced plan, false otherwise.
+	 */
+	public function has_advanced_plan( $refresh_from_wpcom = false ) {
+		static $has_advanced_plan = null;
+		if ( ! $has_advanced_plan ) {
+			$has_advanced_plan = Current_Plan::supports( 'social-image-generator', $refresh_from_wpcom );
```

Comment:
Is there a better way to check the Advanced plan that I should be using instead?

Reply:
I think we could use the instagram plan feature now couldn't we?

Reply:
I've just seen that this is for displaying different notices, but I think the feature check should be for the enhanced publishing feature, as that will allow you to select an image.